### PR TITLE
disable injecting unnecessary variables allowing access to k8s API

### PR DESCRIPTION
**Pull request by paulfantom**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

This will explicitly specify which applications need SA token and which do not need it.

Due to how prometheus server is managed, this PR requires https://github.com/prometheus-operator/prometheus-operator/pull/4514 to be merged first.

This is a part of https://github.com/prometheus-operator/kube-prometheus/issues/1589

## Type of change

_What type of changes does your code introduce to the kube-prometheus? Put an `x` in the box that apply._

- [x] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. Later this will be copied to the changelog file._

```release-note
Specify SA token automounting on pod-level where necessary
```

### CODE REVIEW

File: jsonnet/kube-prometheus/components/prometheus.libsonnet [hunk 1]
```diff
@@ -98,6 +98,7 @@ function(params) {
     apiVersion: 'v1',
     kind: 'ServiceAccount',
     metadata: p._metadata,
+    automountServiceAccountToken: false,
```
Comment (paulfantom) [inline jsonnet/kube-prometheus/components/prometheus.libsonnet:101]: This part can be treated as a breaking change for folks who are using prometheus-operator in versions that do not have change from https://github.com/prometheus-operator/prometheus-operator/pull/4514. Do you think we should wait with this until we have a new operator version and then circle back?

Comment (PhilipGough) [reply jsonnet/kube-prometheus/components/prometheus.libsonnet:101]: yes sounds good

Comment (ArthurSens) [reply jsonnet/kube-prometheus/components/prometheus.libsonnet:101]: Just out of curiosity, what is the next release date?

Comment (paulfantom) [reply jsonnet/kube-prometheus/components/prometheus.libsonnet:101]: Yesterday :smile:

**Comment by paulfantom** [comment]
rebased

**Review by PhilipGough — APPROVED** [review]
lgtm

Aside, if we had some way of generating cert and configure scraping over TLS or mTLS would it be wise to remove (optionally) kube-rbac-proxy or are there some other considerations to be had?

**Review by ArthurSens — APPROVED** [review]
Nice, thanks for taking care of this one!

**Pull request merged**